modyn/protos/trainer_server.proto

```diff
@@ -40,7 +40,7 @@ message StartTrainingRequest {
   JsonString model_configuration = 6;
   bool use_pretrained_model = 7;
   bool load_optimizer_state = 8;
-  string pretrained_model_path = 9;
+  optional int32 pretrained_model_id = 9;
```

Comment:
This is actually something that we also discuss with @francescodeaglio in the seed everything PR:

Apparently, at least for non-optional fields, if they are None, they default to 0. This is why we have the use_pretrained_model boolean.

However, I didn't know about optional fields. The question for me is what the `optional` keyword changes. If the optional keyword prevents fields from defaulting to 0 such that they are None instead of 0 on the receiving end, we can actually remove the `use_pretrained_model` flag. However, if they still default to 0, we still require the bool, but then can remove `optional`. So either bool+no optional or optional + no bool, but not optional + bool. If you (or @francescodeaglio, whoever gets to it first :D) could figure out what `optional` does that'd be great.

Reply:
Okay, yes I agree with this. Removed the optional in latest commit.